dali/pipeline/util/thread_pool.h

```diff
@@ -39,21 +39,14 @@ class DLL_PUBLIC ThreadPool {
   DLL_PUBLIC ~ThreadPool();
 
   /**
-   * @brief Adds work to the queue with optional priority.
-   *        The work only gets queued and it will only start after invoking
-   *        `RunAll` (wakes up all threads to complete all remaining works) or
-   *        `DoWorkWithID` (wakes up a single thread to complete one work unit).
-   * @remarks if finished_adding_work == true, the thread pool will proceed picking
-   *          tasks from its queue, otherwise it will hold execution until `RunAll`
-   *          is invoked.
+   * @brief Adds work to the queue with optional priority, and optionally starts processing
+   *
+   * The jobs are queued but the workers don't pick up the work unless they have
+   * already been started by a previous call to AddWork with start_immediately = true or RunAll.
```

Comment:
```suggestion
   * already been started and not completed by a previous call to AddWork with start_immediately = true or RunAll.
```

Reply:
That's not true.

Reply:
Moreover, that was one of the motivations for this change: so that mere completion does not prevent subsequent jobs from being picked up. The jobs will stop being picked up only after a call to WaitForWork (whereas completion may happen multiple times in between).